liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/ServerXmlDocument.java

```diff
@@ -42,38 +45,84 @@
     
     private ServerXmlDocument() {    
     }
+
+    /**
+     * @param filePath
+     * @param fileName application file name
+     * @return true if the application with the file name is configured.
+     * @throws Exception
+     */
+    private static boolean isFoundWebAppWithFileName(String filePath, String fileName) throws Exception {
+        return isFoundAppConfigByTagAndFileName(filePath, "webApplication", fileName);
+    }
     
-    public static boolean isFoundTagName(String fileName, String tag) throws Exception {
+    /**
+     * @param filePath 
+     * @param tag specify element tag to parse
+     * @param fileName application file name
+     * @return true if the application with the file name within the specified element tag is configured. 
+     * @throws Exception
+     */
+    private static boolean isFoundAppConfigByTagAndFileName(String filePath, String tag, String fileName) throws Exception {
 
-        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
-        DocumentBuilder builder = factory.newDocumentBuilder();
-        Document doc = builder.parse(fileName);
+        NodeList appList = getNodeList(filePath, "/server/" + tag);
+        
+        if (appList == null || appList.getLength() == 0) {
+            return false; 
+        }
+        else {
+          if (fileName == null) {
```

Comment:
Actually, null fileName is an valid argument. If it just needs to check any app configuration is found regardless file name, it passes null. If you don't like using null as a parameter we can use a string constant instead.